beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -635,10 +635,21 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
                     self.log,
                     "Successfully verified gossip data column sidecar";
                     "slot" => %slot,
-                    "root" => %root,
+                    "block_root" => %block_root,
                     "index" => %index,
                 );
 
+                // We have observed a data column sidecar with valid inclusion proof, such that
+                // `block_root` must have data. The column may or not be imported yet.
+                // TODO(das): Sampling should check that sampling is not completed already.
+                //
+                // Trigger sampling early, potentially before processing the block. At this point column
+                // custodials may not have received all their columns. Triggering sampling so early is
+                // only viable with either:
+                // - Sync delaying sampling until some latter window
+                // - Re-processing early sampling requests: https://github.com/sigp/lighthouse/pull/5569
+                self.send_sync_message(SyncMessage::SampleBlock(block_root, slot));
```

Comment:
Minor - for supernodes, we might get flooded with `Received SampleBlock message` if we're sending a sync message for every gossip column received
https://github.com/sigp/lighthouse/blob/f8acdac4c4dc4cb54ebaa5d3871302f0c78bd6c6/beacon_node/network/src/sync/manager.rs#L656

Reply:
Right, triggering on column is a timing optimization. We can drop it for simplicity